### PR TITLE
Homework20: Ekaterina Tsiunel 

### PR DESCRIPTION
**Pull request by ETsiunel**
unittest

### CODE REVIEW

File: hw_20/Library.log
Comment (Stml89) [inline hw_20/Library.log:1]: this is a log file, it shouldn’t be in the repo

File: hw_20/bank.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        deposit = Deposit(N, R)
+        return deposit.total_amount()
+
+
+bank = Bank()
+N = 1000  # сумма вклада
+R = 12  # срок вклада в месяцах
+total_amount = bank.deposit(N, R)
+
+print(f"Сумма на счету через {R} месяцев: {total_amount:.4f} рублей")
```
Comment (Stml89) [inline hw_20/bank.py]: if you do not want to execute some code when module is imported, use `if __name__ == "__main__"`

File: hw_20/testBank.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import logging
+from hw_20.bank import Deposit, Bank
+
+formatter = logging.Formatter('\n[%(asctime)s] - %(levelname)s - %(message)s')
+handler = logging.StreamHandler()
+handler.setFormatter(formatter)
+
+logger = logging.getLogger(__name__)
+logger.addHandler(handler)
+logger.setLevel(logging.INFO)
```
Comment (Stml89) [inline hw_20/testBank.py]: You can use 1 logger across single application, e.g.: https://stackoverflow.com/questions/75489656/how-to-use-a-single-python-logging-object-throughout-the-python-application

File: hw_20/testBank.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        logger.info(f'Expected amount = {expected_amount:.4f}, '
+                    f'Calculated amount = {calculated_amount:.4f}')
+
+        # N = 2000 RUB, R = 6 month, percent = 5%
+        logger.info('Testing total amount: initial 2000 RUB, 6 months, 5%')
+        expected_amount = 2000 * ((1 + 0.05 / 12) ** 6)
+        calculated_amount = self.deposit3.total_amount()
+        self.assertAlmostEqual(calculated_amount, expected_amount, places=2)
+        logger.info(f'Expected amount = {expected_amount:.4f}, '
+                    f'Calculated amount = {calculated_amount:.4f}')
```
Comment (Stml89) [inline hw_20/testBank.py]: Looks like - 3 different test cases. But those 3 TCs are similar and verify the same functionality, but you can use other TCs: negative present rate, really long duration 10y and etc.

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: hw_20/Library.log
Comment (ETsiunel) [reply hw_20/Library.log:1]: fixed

File: hw_20/bank.py (hunk 1 above)
Comment (ETsiunel) [reply hw_20/bank.py]: done

File: hw_20/testBank.py (hunk 2 above)
Comment (ETsiunel) [reply hw_20/testBank.py]: done

File: hw_20/testBank.py (hunk 3 above)
Comment (ETsiunel) [reply hw_20/testBank.py:56]: done

Comment (ETsiunel) [reply hw_20/testBank.py]: will be done

Comment (ETsiunel) [reply hw_20/testBank.py]: done

**Comment by ETsiunel** [comment]
ready for review

File: hw_20/library.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+user1.reserve_book(book1)
+user2.take_book(book1)
+user1.take_book(book1)
+user2.take_book(book2)
+user2.return_book(book1)
+user1.return_book(book1)
+user2.reserve_book(book1)
+user2.take_book(book1)
+user1.take_book(book2)
```
Comment (Stml89) [inline hw_20/library.py]: The same: `if __name__ == "__main__"`

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: hw_20/library.py (hunk 4 above)
Comment (ETsiunel) [reply hw_20/library.py]: done

**Comment by ETsiunel** [comment]
all changes pushed and ready for review

**Review by Stml89 — APPROVED** [review]

**Pull request merged**